### ISSUE TITLE
torch.compile, when passes a single torch operation, does not put it into a graph

### ISSUE THREAD

**Issue by v0i0** [pytorch/pytorch#168284]
### 🐛 Describe the bug

Running these two pieces of code:

```
a = torch.compile(torch.exp)(x)
b = torch.compile(lambda x: torch.exp(x))(x)
```

behave differently. The first will just call the python operation, the second will capture it in a graph and go through the compile stack.

This is surprising because there is not reason, from a user perspective, for these to be different.
I might for example compile individual ops to opt into gemm autotuning, or for accuracy checking.

It is extra confusing if passing `fullgraph=True`, where we'd expect that either all pytorch operations are captured into a single graph, or we receive an error.

### Versions

nightly

cc @chauhang @penguinwu @voznesenskym @EikanWang @jgong5 @Guobing-Chen @XiaobingSuper @zhuhaozhe @blzheng @wenzhe-nrv @jiayisunx @kadeng @amjames @Lucaskabela @jataylo @chenyang78

**Comment by masnesral** [comment]
I would assume it's because torch.exp is probably a C function (<class 'builtin_function_or_method'>)? Perhaps dynamo only wants to trace user-defined functions and nn modules? I'd need someone from dynamo to give the real answer. Tagging with module: dynamo

**Comment by manigkrish** [comment]
Got it, thanks!

I’ll first send a PR for the top-level `TorchInGraphFunctionVariable` behavior (single-op functions like `torch.exp` producing a 1-node FX graph), and then follow up with a separate PR for the `fullgraph=True` case that errors when no FX graph is traced.
